### PR TITLE
testing simple offload engine

### PR DESCRIPTION
**Pull request by arjunrawal4**
cost model file is work in progress but provides a starting point

### CODE REVIEW

File: pandasql/core.py [hunk 1]
```diff
@@ -353,8 +357,9 @@ def __init__(self, data=None, name=None, sources=None, deep_copy=False):
 
         if df is not None and len(df) > 0:
             # Offload dataframe to SQLite
-            df.to_sql(name=self.name, con=SQL_CON, index=False)
+            df.to_sql(name=self.name, con=SQL_CON, index=False, chunksize=10000)
             self._cached_on_sqlite = True
+            self._input_size = self._cached_result.memory_usage(deep=True).sum()
```
Comment (rohankumar42) [inline pandasql/core.py]: We should update `self._input_size` when the result is computed too, e.g., at the end of `compute`

**Review by rohankumar42 — CHANGES_REQUESTED** [review]

File: pandasql/core.py (hunk 1 above)
Comment (arjunrawal4) [reply pandasql/core.py]: I'm not 100% sure what the best thing to do here is. I guess it depends on how we are using `input_size`. Right now if we updated it after `compute`, the current cost model would count both the original `input_size` and the new `input_size` from the cached result.

File: pandasql/core.py [hunk 2]
```diff
@@ -111,6 +112,13 @@ def _compute_sqlite(self):
 
         return self.result
 
+    def memory_usage(self):
+        if isinstance(self._cached_result, pd.DataFrame):
+            return self._cached_result.memory_usage(deep=True, index=True).sum()
+        elif isinstance(self._cached_result, pd.Series):
+            return self._cached_result.memory_usage(deep=True, index=True)
+        # TODO: handle Aggregator, GroupByDataFrame, and GroupByProjection
```
Comment (rohankumar42) [inline pandasql/core.py]: Should this be returning something in the default case?

File: pandasql/core.py [hunk 3]
```diff
@@ -62,6 +62,7 @@ def compute(self):
                 self._compute_sqlite()
             else:
                 self._compute_pandas()
+        self._input_size = self.memory_usage()
```
Comment (rohankumar42) [inline pandasql/core.py]: It's a little weird to have both a per-object `memory_usage` function and a `_input_size` variable that stores its result. Maybe any code that needs the memory usage should just call `memory_usage`?

File: pandasql/core.py (hunk 2 above)
Comment (arjunrawal4) [reply pandasql/core.py]: I think None should be easiest, as it indicates we don't know, as opposed to 0 or something else.

Comment (rohankumar42) [reply pandasql/core.py]: Would a `NotImplemented` be useful?

File: pandasql/core.py (hunk 3 above)
Comment (arjunrawal4) [reply pandasql/core.py]: I included this because the pd.memory_usage call can be slow, so it seems like it might be worth it to cache the result

Comment (rohankumar42) [reply pandasql/core.py]: I'd say remove the `_input_size` for now. We can (either now or later) ensure that the memory usage computation is just done once.

**Pull request merged**